iroh-bitswap/src/lib.rs

```diff
@@ -216,16 +237,22 @@ impl<S: Store> Bitswap<S> {
 
     pub async fn stop(self) -> Result<()> {
         self.network.stop();
-        let (a, b) = futures::future::join(self.client.stop(), self.server.stop()).await;
-        a?;
-        b?;
+        if let Some(server) = self.server {
+            let (a, b) = futures::future::join(self.client.stop(), server.stop()).await;
+            a?;
+            b?;
```

Comment:
How about [try_join](https://docs.rs/futures/latest/futures/future/fn.try_join.html) here?

```suggestion
            futures::future::try_join(self.client.stop(), server.stop()).await?;
```